src/core/prism-class.ts

```diff
@@ -0,0 +1,49 @@
+import { Hooks } from './hooks';
+import { Registry } from './registry';
+import { highlightAll, type HighlightAllOptions } from './highlight-all';
+import { highlightElement, type HighlightElementOptions } from './highlight-element';
+import { highlight, type HighlightOptions } from './highlight';
+import { tokenize } from './tokenize';
+import type { KnownPlugins } from '../known-plugins';
+import type { Grammar, GrammarToken, GrammarTokens, RegExpLike } from '../types';
+import type { TokenStream } from './token';
+
+/**
+ * Prism: Lightweight, robust, elegant syntax highlighting
+ *
+ * @license MIT <https://opensource.org/licenses/MIT>
+ * @author Lea Verou <https://lea.verou.me> and contributors <https://github.com/PrismJS/prism/graphs/contributors>
+ */
+export default class Prism {
+	hooks = new Hooks();
+	components = new Registry(this);
```

Comment:
It looks like we have only languages there (plugins have their own property). Why not use languages instead of components here?

Reply:
Agreed. Can you do it?

Reply:
Sure, will do